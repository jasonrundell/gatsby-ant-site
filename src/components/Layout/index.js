import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { Link, navigate } from 'gatsby'
import { Layout, Menu, TreeSelect } from 'antd'

import SkipToMain from '../SkipToMain'
import Breadcrumb from '../Breadcrumb'

import styles from './index.module.scss'

import 'antd/dist/antd.css'

const { Header, Content, Footer } = Layout
const { TreeNode } = TreeSelect

const _Layout = ({ title, crumbs, children, pathname }) => {
  const [menuValue, setMenuValue] = useState(undefined)

  const onChange = value => {
    navigate(value)
  }
  return (
    <>
      <Header className={styles.header}>
        <Link to="/" className={styles.siteName}>
          {title}
        </Link>
        <Menu
          className={styles.visibleLarge}
          theme="dark"
          mode="horizontal"
          selectedKeys={[pathname]}
          style={{ lineHeight: '4rem' }}
        >
          <Menu.Item key="/">
            <Link to="/">Home</Link>
          </Menu.Item>
          <Menu.Item key="/blog/">
            <Link to="/blog/">Blog</Link>
          </Menu.Item>
          <Menu.Item key="/products/">
            <Link to="/products/">Products</Link>
          </Menu.Item>
          <Menu.Item key="/contact-us/">
            <Link to="/contact-us/">Contact Us</Link>
          </Menu.Item>
        </Menu>
        <div className={styles.mobileMenu}>
          <TreeSelect
            showSearch
            value={menuValue}
            dropdownStyle={{ maxHeight: 400, overflow: 'auto' }}
            placeholder="Menu"
            allowClear
            treeDefaultExpandAll
            onChange={onChange}
          >
            <TreeNode value="/" title="Home" key="/" />
            <TreeNode value="/blog/" title="Blog" key="/blog/" />
            <TreeNode value="/products/" title="Products" key="/products/" />
            <TreeNode
              value="/contact-us/"
              title="Contact Us"
              key="/contact-us/"
            />
          </TreeSelect>
        </div>
      </Header>
      <SkipToMain />
      <Content id="main" className={styles.content}>
        {crumbs && <Breadcrumb crumbs={crumbs} />}
        {children}
      </Content>
      <Footer className={styles.footer}>
        Â© {new Date().getFullYear()}. Built with
        {` `}
        <a href="https://www.gatsbyjs.org" rel="noopener noreferrer">
          Gatsby
        </a>{' '}
        and{' '}
        <a href="https://ant.design/" rel="noopener noreferrer">
          Ant Design
        </a>
        .{` `}
      </Footer>
    </>
  )
}

_Layout.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
}

export default _Layout
