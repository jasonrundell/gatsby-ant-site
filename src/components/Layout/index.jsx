import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'gatsby'
import { Layout } from 'antd'

import SkipToMain from '../SkipToMain'
import Breadcrumb from '../Breadcrumb'
import MainNav from '../MainNav'

import 'antd/dist/antd.css'
import '../../styles/antd-overrides.scss'

import styles from './Layout.module.scss'

const { Header, Content, Footer } = Layout

const _Layout = ({ title, crumbs, children, pathname }) => {
  return (
    <>
      <Header className={styles.header}>
        <Link to="/" className={styles.siteName}>
          {title}
        </Link>
        <MainNav pathname={pathname} />
      </Header>
      <SkipToMain />
      <Content id="main" className={styles.content}>
        {crumbs && <Breadcrumb crumbs={crumbs} />}
        {children}
      </Content>
      <Footer className={styles.footer}>
        Â© {new Date().getFullYear()}. Built with
        {` `}
        <a href="https://www.gatsbyjs.org" rel="noopener noreferrer">
          Gatsby
        </a>{' '}
        and{' '}
        <a href="https://ant.design/" rel="noopener noreferrer">
          Ant Design
        </a>
        .{` `}
      </Footer>
    </>
  )
}

_Layout.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
}

export default _Layout
