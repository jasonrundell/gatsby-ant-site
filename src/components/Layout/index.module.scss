@import '../../styles/theme';

.header {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 1;
  width: 100%;
}

.content {
  margin-top: 4rem;
  margin-bottom: 4rem;
  padding-top: 1rem;
  padding-right: 2rem;
  padding-bottom: 1rem;
  padding-left: 2rem;
}

.site-name {
  float: left;
  margin-right: 1.5rem;
  font-size: 1.5rem;
  font-weight: 400;
  color: #fff;
  &:visited {
    color: #fff;
  }
}

.footer {
  position: fixed;
  bottom: 0;
  left: 0;
  z-index: 1;
  width: 100%;
}
