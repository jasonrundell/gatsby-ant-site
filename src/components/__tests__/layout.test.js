import React from 'react'
import renderer from 'react-test-renderer'

import Layout from '../layout'

describe('Layout', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Layout location="/blog/" title="Layout title">
          Layout test
        </Layout>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
