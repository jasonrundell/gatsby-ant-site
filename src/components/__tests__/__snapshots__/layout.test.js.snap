// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <h1
        className="title"
      >
        <a
          className="link"
          href="/"
        >
          Layout title
        </a>
      </h1>
    </div>
  </header>,
  <div
    className="container"
  >
    <nav
      aria-label="Breadcrumb"
      className="breadcrumb"
    >
      <ol>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/blog/"
          >
            Blog
          </a>
        </li>
      </ol>
    </nav>
  </div>,
  <div
    className="container"
  >
    <main>
      Layout test
    </main>
    <footer
      className="footer"
    >
      Â© 
      2019
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
        rel="noopener noreferrer"
      >
        Gatsby
      </a>
      .
       
      Calin says hello.
    </footer>
  </div>,
]
`;
